zookeeper-server/src/main/java/org/apache/zookeeper/backup/BackupFileInfo.java

```diff
@@ -0,0 +1,124 @@
+package org.apache.zookeeper.backup;
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.io.File;
+import com.google.common.collect.Range;
+import org.apache.zookeeper.backup.BackupUtil.BackupFileType;
+import org.apache.zookeeper.backup.BackupUtil.ZxidPart;
+import org.apache.zookeeper.server.persistence.Util;
+
+public class BackupFileInfo {
+  private final File backupFile;
+  private final File standardFile;
+  private final Range<Long> zxidRange;
+  private final BackupFileType fileType;
+  private final long modificationTime;
+  private final long size;
+
+  /**
+   * Constructor that pulls backup metadata based on the backed-up filename
+   * @param backedupFile the backed-up file with the name in the form prefix.lowzxid-highzxid
+   *                     for example snapshot.9a0000a344-9a0000b012
+   * @param modificationTime the file modification time
+   * @param size the size of the file in bytes
+   */
+  public BackupFileInfo(File backedupFile, long modificationTime, long size) {
+    this.backupFile = backedupFile;
+    this.modificationTime = modificationTime;
+    this.size = size;
+
+    String backedupFilename = this.backupFile.getName();
+
+    if (backedupFilename.startsWith(BackupUtil.LOST_LOG_PREFIX)) {
+      this.fileType = BackupFileType.LOSTLOG;
+      this.standardFile = this.backupFile;
+    } else if (backedupFilename.startsWith(Util.SNAP_PREFIX)) {
+      this.fileType = BackupFileType.SNAPSHOT;
+      this.standardFile = new File(this.backupFile.getParentFile(), backedupFilename.split("-")[0]);
+    } else if (backedupFilename.startsWith(Util.TXLOG_PREFIX)) {
+      this.fileType = BackupFileType.TXNLOG;
+      this.standardFile = new File(this.backupFile.getParentFile(), backedupFilename.split("-")[0]);
+    } else {
+      throw new IllegalArgumentException("Not a known backup file type: " + backedupFilename);
+    }
```

Comment:
This could be a general method, right? It could help to back tracing the standard name to backup file name. Why not let's create a util method for that? Then in the future, if we would like to do some look up, we can reuse the code.

Reply:
Thanks for the input. However, this whole class is a remnant of the OSS patch I believe, so although we could consider revising, I would leave this logic as is to keep the difference at a minimum.